### PR TITLE
prim disc problem

### PR DESCRIPTION
**Pull request by MaHmOuDeB**
I know where the problem comes from
The first part of the intersection algorithm creates a plane, and then the second part creates the disc
When the texture gets shifted, then the points that were before part of the disc, are not anymore part of it, so we get this texture shifting

### CODE REVIEW

File: modules/core/PrimDisc.cpp [hunk 1]
```diff
@@ -7,11 +7,13 @@ namespace rt {
 	{
 		float dist = (m_origin - ray.org).dot(m_normal) / ray.dir.dot(m_normal);
 		if (dist < Epsilon || isinf(dist) || dist > ray.t) return false;
-		if (static_cast<float>(norm(ray.org + ray.dir * dist - m_origin)) > m_radius) return false;
+		if (static_cast<float>(norm(ray.org + ray.dir * dist - m_origin)) <= m_radius) return false;
```
Comment (ereator) [inline modules/core/PrimDisc.cpp]: I think this is an error, should be `> m_radius`

File: modules/core/PrimDisc.cpp [hunk 2]
```diff
@@ -7,11 +7,13 @@ namespace rt {
...
 
 		ray.t = dist;
 		ray.hit = shared_from_this();
 		return true;
+
```
Comment (ereator) [inline modules/core/PrimDisc.cpp:15]: Please remove these 2 redundant lines

File: modules/core/PrimDisc.h [hunk 3]
```diff
@@ -26,6 +26,7 @@ namespace rt {
 			, m_normal(normal)
 			, m_origin(origin)
 			, m_radius(radius)
+               , m_t(Mat::eye(4, 4, CV_32FC1))
 		{}
```
Comment (ereator) [inline modules/core/PrimDisc.h]: Another error, you did not initialize your pivot point. You need to add the line  
`for (int i = 0; i < 3; i++) m_t.at<float>(i, 3) = m_origin[i];` 
in constructor

**Review by ereator — CHANGES_REQUESTED** [review]
The reason why it does not work is that you forgot to copy also the initialisation of the _pivot point_ from Sphere's constructor.

File: modules/core/PrimDisc.h (hunk 3 above)
Comment (MaHmOuDeB) [reply modules/core/PrimDisc.h]: I understand, I felt that something wasn't making sense, thank you Professor!

File: modules/core/PrimDisc.cpp (hunk 1 above)
Comment (MaHmOuDeB) [reply modules/core/PrimDisc.cpp]: Right, I was debugging and I forgot to put it back the way it should be

**Pull request closed without merging**